tls/s2n_client_hello.c

```diff
@@ -917,6 +920,34 @@ int s2n_client_hello_get_session_id(struct s2n_client_hello *ch, uint8_t *out, u
     return S2N_SUCCESS;
 }
 
+int s2n_client_hello_get_compression_methods_length(struct s2n_client_hello *ch, uint32_t *out_length)
+{
+    POSIX_ENSURE_REF(ch);
+    POSIX_ENSURE_REF(out_length);
+    *out_length = ch->compression_methods.size;
+    return S2N_SUCCESS;
+}
+
+int s2n_client_hello_get_compression_methods(struct s2n_client_hello *ch, uint8_t *out, uint32_t *out_length, uint32_t max_length)
+{
+    POSIX_ENSURE_REF(ch);
+    POSIX_ENSURE_REF(out);
+    POSIX_ENSURE_REF(out_length);
+
+    uint32_t len = min_size(&ch->compression_methods, max_length);
```

Comment:
With other getters we error if the provided max length was too small. Was it intentional to just write out as much as possible instead of erroring? Is there a concern that the user might not know they aren't getting all of the compression methods?